neuralpredictors/layers/readouts/gaussian.py

```diff
@@ -277,13 +277,26 @@ def __init__(
         feature_reg_weight=None,
         gamma_readout=None,  # depricated, use feature_reg_weight instead
         return_weighted_features=False,
+        regularizer_type="l1",
+        gamma_sigma=0.1,
         **kwargs,
     ) -> None:
         super().__init__()
         self.feature_reg_weight = self.resolve_deprecated_gamma_readout(feature_reg_weight, gamma_readout, default=1.0)
         self.mean_activity = mean_activity
         # determines whether the Gaussian is isotropic or not
         self.gauss_type = gauss_type
+        self._regularizer_type = regularizer_type
+
+        if self._regularizer_type == "adaptive_log_norm":
+            self.gamma_sigma = gamma_sigma
+            self.adaptive_neuron_reg_coefs = torch.nn.Parameter(
+                torch.normal(mean=torch.ones(1, outdims), std=torch.ones(1, outdims))
+            )
+        elif self._regularizer_type != "l1":
+            raise ValueError(
+                f"_regularizer_type should be 'l1' or 'adaptive_log_norm' but got {self._regularizer_type}"
```

Comment:
```suggestion
                f"regularizer_type should be 'l1' or 'adaptive_log_norm' but got {self._regularizer_type}"
```
(because it refers to the parameter passed at calling `__init__`)

Reply:
(i.e. you as a user cannot edit the protected attribute in this context, but only the argument you pass)